components/feature/sendtab/src/main/java/mozilla/components/feature/sendtab/SendTabFeature.kt

```diff
@@ -0,0 +1,150 @@
+/*
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ *  License, v. 2.0. If a copy of the MPL was not distributed with this
+ *  file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+package mozilla.components.feature.sendtab
+
+import android.content.Context
+import androidx.lifecycle.LifecycleOwner
+import androidx.lifecycle.ProcessLifecycleOwner
+import mozilla.components.concept.push.Bus
+import mozilla.components.concept.push.PushService
+import mozilla.components.concept.sync.AccountObserver as SyncAccountObserver
+import mozilla.components.concept.sync.Device
+import mozilla.components.concept.sync.DeviceConstellation
+import mozilla.components.concept.sync.DeviceEvent
+import mozilla.components.concept.sync.DeviceEventsObserver
+import mozilla.components.concept.sync.DevicePushSubscription
+import mozilla.components.concept.sync.OAuthAccount
+import mozilla.components.concept.sync.TabData
+import mozilla.components.feature.push.AutoPushFeature
+import mozilla.components.feature.push.AutoPushSubscription
+import mozilla.components.feature.push.PushSubscriptionObserver
+import mozilla.components.feature.push.PushType
+import mozilla.components.service.fxa.manager.FxaAccountManager
+import mozilla.components.support.base.log.logger.Logger
+
+/**
+ * A feature that uses the [FxaAccountManager] to send and receive tabs with optional push support
+ * for receiving tabs from the [AutoPushFeature] and a [PushService].
+ *
+ * If the push components are not used, the feature can still function while tabs would only be
+ * received when the account is next synced.
+ *
+ * @param context Android application context for starting the (optional) push service.
+ * @param accountManager Firefox account manager.
+ * @param pushFeature The [AutoPushFeature] if that is setup for observing push events.
+ * @param pushService The optional [PushService] to manage start/stop of the service when
+ * an account is signed in/out.
+ * @param owner Android lifecycle owner for the observers. Defaults to the [ProcessLifecycleOwner]
+ * so that we can always observe events throughout the application lifecycle.
+ * @param autoPause whether or not the observer should automatically be
+ * paused/resumed with the bound lifecycle.
+ * @param onTabsReceived the callback invoked with new tab(s) are received.
+ */
+class SendTabFeature(
+    context: Context,
+    accountManager: FxaAccountManager,
+    pushFeature: AutoPushFeature? = null,
+    pushService: PushService? = null,
```

Comment:
IIUC, you use `pushService` below as part of the `AccountObserver`. But, `AutoPushFeature` already has a `shutdown` and `initialize` methods for logging out and starting, which do what you want. Why not just delegate this work to the `pushFeature`, and remove `pushService` as a parameter here?

Reply:
The `pushService` definitely doesn't need to be there. I added in the docs that this service can be passed in if you want the service to be managed by the feature to be turned on/off when an account is signed in/out.

> Why not just delegate this work to the pushFeature, and remove pushService as a parameter here?

Correct, `AutoPushFeature.initialize` does start the service and we could just use the feature to do that, however the `AutoPushFeature.shutdown` doesn't just shutdown the service, but also unsubscribes all the subscriptions. If we want to remove the optional service from being managed by the feature, I don't think it's necessary to make the AutoPushFeature even start/stop the service.